backends/dpdk/dpdkArch.cpp

```diff
@@ -1666,6 +1666,16 @@ const IR::Node *CopyMatchKeysToSingleStruct::postorder(IR::KeyElement *element)
     }
 
     auto keyName = element->expression->toString();
+    if (auto m = element->expression->to<IR::MethodCallExpression>()) {
+        /* Moving <hdr>.isValid() used as table key to Metadata */
```

Comment:
Why not just extract the path expression or member from the method call and then convert it into a string? For me that is the safest option because you are directly accessing the path.

Reply:
Also check that the expression is a method call before calling `toString`.

Reply:
The table keys at this point could be PathExpression , Member or MethodCallExpression (only isValid). 
The first toString() call on element->expression handles all cases and later isValid is handled. I added a check for confirming that the method call is indeed an isValid call and based on that the keyName is obtained now.

Reply:
What I mean to say is that you can implement this without ever needing to call `toString`. Instead, you can check that each type is either a member, a path expression, or a valid method call. Then resolve the expression appropriately into its subparts. That is how it is done in other parts of the compiler. This way you safely enforce typing, too. 

Largely because `toString` is not meant for the purpose it is used here. The semantics of the function can change, like it happened before.